### PR TITLE
Tomcat connectors

### LINKED ISSUE #269: Jira 7.12 needs relaxedPathChars & relaxedQueryChars
Jira 7.12 runs on tomcat 8.5 and needs additional connector parameters -

https://confluence.atlassian.com/jirakb/changing-server-xml-to-handle-requests-with-special-characters-958453799.html

Here's the new default connector -

```
<Connector
  port="8080"
  relaxedPathChars="[]|"
  relaxedQueryChars="[]|{}^&#x5c;&#x60;&quot;&lt;&gt;"
  maxThreads="150"
  minSpareThreads="25"
  connectionTimeout="20000"
  enableLookups="false"
  maxHttpHeaderSize="8192"
  protocol="HTTP/1.1"
  useBodyEncodingForURI="true"
  redirectPort="8443"
  acceptCount="100"
  disableUploadTimeout="true"
  bindOnInit="false"
/>
```

Please provide a way to add relaxedPathChars and relaxedQueryChars.  Thanks!

### PR DESCRIPTION
**Pull request by balinteu**

### CODE REVIEW

File: templates/server.xml.erb [hunk 1]
```diff
@@ -46,6 +46,10 @@
 
     <Service name="Catalina">
         <Connector port="<%= @tomcat_port %>"
+                   <%- if Gem::Version.new(@version) >= Gem::Version.new('7.12.0') -%>
+                   relaxedPathChars="[]|" 
+                   relaxedQueryChars="[]|{}^&#x5c;&#x60;&quot;&lt;&gt;"
```
Comment (bastelfreak) [inline templates/server.xml.erb]: are there some strange characters after `}`?

**Comment by bastelfreak** [comment]
Hi @balinteu. I closed and reopened the PR to kick of travis. Can you check the template? It has some strange characters in it.

File: templates/server.xml.erb [hunk 2]
```diff
@@ -71,19 +75,22 @@
...
-                    SSLEnabled="true"
-                    maxThreads="<%= @tomcat_max_threads %>"
-                    minSpareThreads="<%= @tomcat_min_spare_threads %>"
-                    enableLookups="<%= @tomcat_enable_lookups %>"
-                    disableUploadTimeout="<%= @tomcat_disable_upload_timeout %>"
-                    acceptCount="<%= @tomcat_accept_count %>"
+        <Connector port="<%= @tomcat_https_port %>"
+                   <%- if Gem::Version.new(@version) >= Gem::Version.new('7.12.0') -%>
+                   relaxedPathChars="[]|" 
+                   relaxedQueryChars="[]|{}^&#x5c;&#x60;&quot;&lt;&gt;"
```
Comment (bastelfreak) [inline templates/server.xml.erb]: same here?

Comment (balinteu) [reply templates/server.xml.erb]: Hi @bastelfreak. Those are the entries specified in the Atlassian upgrade notes: 
https://confluence.atlassian.com/jirasoftware/jira-software-7-12-x-upgrade-notes-955173978.html
More precisely: "Add relaxedPathChars="[]|" relaxedQueryChars="[]|{}^&#x5c;&#x60;&quot;&lt;&gt;" to the connector properties in server.xml"

**Comment by fabiogermann** [comment]
Is this PR still needed now that pull request #288 has been merged?

**Comment by mojibake-umd** [comment]
Can be closed? Added in f6bd5405ec50ae5e444a38a869b7b647d9764ff9

**Comment by ekohl** [comment]
Closing per https://github.com/voxpupuli/puppet-jira/pull/277#issuecomment-554998543.

**Pull request closed without merging**